### PR TITLE
KAFKA-15859: Complete delayed RemoteListOffsets requests when replica moves away from broker

### PR DESCRIPTION
**Pull request by kamalcph**
- Removed the ListOffsetsMetadata wrapper class.
- Complete the response when replica is not a leader or follower.
- Addressed review comments from PR #16602.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/server/DelayedRemoteListOffsets.scala [hunk 1]
```diff
@@ -31,7 +31,7 @@ import java.util.concurrent.TimeUnit
 import scala.collection.{Map, mutable}
 import scala.jdk.CollectionConverters._
 
-case class ListOffsetsPartitionStatus(var responseOpt: Option[ListOffsetsPartitionResponse] = None,
+case class ListOffsetsPartitionStatus(@volatile var responseOpt: Option[ListOffsetsPartitionResponse],
```
Comment (kamalcph) [inline core/src/main/scala/kafka/server/DelayedRemoteListOffsets.scala]: Intellij shows "Class parameter shadows superclass var". How to fix this warning?

Screenshot 2024-10-13 at 1 14 39 AM

Comment (junrao) [reply core/src/main/scala/kafka/server/DelayedRemoteListOffsets.scala]: Probably this conflicts with the case class. Maybe use a non-case class?

Comment (kamalcph) [reply core/src/main/scala/kafka/server/DelayedRemoteListOffsets.scala]: Used a non-case class with object to overcome the warning.

**Review by junrao — APPROVED** [review]
@kamalcph : Thanks for the updated PR. LGTM

**Pull request merged**